examples/threadExample/src/main/java/io/realm/examples/threads/AsyncQueryFragment.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.threads;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import java.util.Collections;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmQuery;
+import io.realm.RealmResults;
+import io.realm.examples.threads.model.Dot;
+
+/**
+ * This fragment demonstrates how you can perform asynchronous queries with Realm
+ */
+public class AsyncQueryFragment extends Fragment implements View.OnClickListener {
+    private Realm realm;
+    private Button mBtnStart;
+    private DotAdapter mAdapter;
+
+    private RealmQuery.Request mAsyncRequest;
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_async_query, container, false);
+        mBtnStart = (Button) rootView.findViewById(R.id.start_button);
+        mBtnStart.setOnClickListener(this);
+
+        ListView listView = (ListView) rootView.findViewById(android.R.id.list);
+        mAdapter = new DotAdapter(getActivity());
+        listView.setAdapter(mAdapter);
+        return rootView;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Create Realm instance for the UI thread
+        realm = Realm.getDefaultInstance();
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        // Remember to close the Realm instance when done with it.
+        realm.close();
+        cancelRequest();
```

Comment:
isn't it dangerous to cancel after you close. If the cancel doesn't work, the callback will succeed and you will crash there because the data is no longer accessible?